src/main/java/com/KacFlor/ShopSpring/service/UserService.java

```diff
@@ -17,7 +21,68 @@ public UserService(UserRepository userRepository){
         this.userRepository = userRepository;
     }
 
-    public List<User> getUser(){
+    public List<User> getAllUser(){
         return userRepository.findAll();
     }
+
+    public User getCurrentUser() {
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String username = ((UserDetails) principal).getUsername();
+
+        return userRepository.findByLogin(username);
+    }
+
+
+    public Boolean changeName(String newName){
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String username = ((UserDetails) principal).getUsername();
+
+
+        User user = userRepository.findByLogin(username);
+        user.setLogin(newName);
+
+        userRepository.save(user);
```

Comment:
It will update this user right? Not add new one?

Reply:
yes i checked it in postman and it changed name